src/main/java/fr/quatrevieux/araknemu/game/handler/chat/SendSmileyToFight.java

```diff
@@ -19,21 +19,34 @@
 
 package fr.quatrevieux.araknemu.game.handler.chat;
 
+import fr.quatrevieux.araknemu.core.network.exception.ErrorPacket;
 import fr.quatrevieux.araknemu.core.network.parser.PacketHandler;
+import fr.quatrevieux.araknemu.game.GameConfiguration;
 import fr.quatrevieux.araknemu.game.fight.fighter.player.PlayerFighter;
 import fr.quatrevieux.araknemu.network.game.GameSession;
 import fr.quatrevieux.araknemu.network.game.in.chat.UseSmiley;
 import fr.quatrevieux.araknemu.network.game.out.chat.Smiley;
+import fr.quatrevieux.araknemu.network.out.ServerMessage;
 import org.checkerframework.checker.nullness.util.NullnessUtil;
 
 /**
  * Send the player smiley to the fight
  */
 public final class SendSmileyToFight implements PacketHandler<GameSession, UseSmiley> {
+    private final GameConfiguration.ChatConfiguration configuration;
+
+    public SendSmileyToFight(GameConfiguration.ChatConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
     @Override
     public void handle(GameSession session, UseSmiley packet) throws Exception {
         final PlayerFighter fighter = NullnessUtil.castNonNull(session.fighter());
 
+        if (!session.get(SpamCheckAttachment.KEY).check(configuration.spamCheckInterval(), configuration.spamCheckMaxCount())) {
```

Comment:
Seems weird that SendSmileyToFight knows the configuration..?
Clearly should be passed to a SpamCheckAttachmentFactory that's passed to sessions

Reply:
You're absolutely right